### PR TITLE
Atq trans charges squash

### PR DESCRIPTION
**Pull request by thomas-niehaus**
Replaces older ATQ pull request to reintroduce ATQ.DAT with fewer commits . Up to date with main.

### CODE REVIEW

File: src/dftbp/timedep/linrespgrad.F90 [hunk 1]
```diff
@@ -5222,4 +5231,87 @@ subroutine conicalIntersectionOptimizer(derivs, excDerivs, indNACouplings, energ
...
+    type(TFileDescr), intent(in) :: fdTagged
+
+    !> Tagged writer
+    type(TTaggedWriter), intent(inout) :: taggedWriter
+
+    integer :: i, ia
+    real(dp), allocatable :: atomicTransQ(:), qia(:)
+    real(dp) :: preFactor
+
+    @:ASSERT(size(xpy(:, lr%nstat) == rpa%nxov_rd))
```
Comment (bhourahine) [inline src/dftbp/timedep/linrespgrad.F90:5281]:
```suggestion
    @:ASSERT(size(xpy(:, lr%nstat)) == rpa%nxov_rd)
```

File: doc/dftb+/manual/dftbp.tex [hunk 2]
```diff
@@ -5524,15 +5523,18 @@ \subsection{Casida}
     created, which contains the spectrum at the uncoupled DFTB level (i.e.\ the
     single-particle excitations).
 
-  \item[\is{WriteTransitionCharges}] For range separated calculations
-    (p.~\pref{sec:dftbp.Hybrid}) setting this to \is{Yes} generates the file
-    ATQ.DAT containing the Mulliken charges for a specific transition (chosen by
-    \is{StateOfInterest}).
+  \item[\is{WriteTransitionCharges}] For singlet or spin-polarized
+    calculations setting this to \is{Yes} generates the file
+    ATQ.DAT containing the Mulliken charges for a specific excited state (chosen by
```
Comment (bhourahine) [inline doc/dftb+/manual/dftbp.tex:5528]: Is "transition" less ambiguous than "excited state", as this isn't an actual excited state set of charges (or did I misunderstand)?

File: doc/dftb+/manual/dftbp.tex [hunk 3]
```diff
@@ -5524,15 +5523,18 @@ \subsection{Casida}
...
+    \is{StateOfInterest}) according to \cite{plotz2014new}.
```
Comment (bhourahine) [inline doc/dftb+/manual/dftbp.tex:5529]:
```suggestion
    \is{StateOfInterest}) according to~\cite{plotz2014new}.
```

File: doc/dftb+/manual/dftbp.tex [hunk 4]
```diff
@@ -5524,15 +5523,18 @@ \subsection{Casida}
...
 
   \item[\is{WriteMulliken}] If set to \is{Yes}, the files XCH.DAT and XREST.DAT
     are created. The former contains atom-resolved Mulliken (gross) charges for
     the excited state of interest, the latter the excited-state dipole moment of
-    the state.
+    the state. Warning: These quantities are computed from the relaxed
+    transition density added to the ground state density and do
```
Comment (bhourahine) [inline doc/dftb+/manual/dftbp.tex:5535]:
```suggestion
    transition density added to the ground state density and possibly do not
```

File: doc/dftb+/manual/dftbp.tex [hunk 5]
```diff
@@ -5524,15 +5523,18 @@ \subsection{Casida}
...
+    possibly not faithfully represent the charges for a given excited many-body
```
Comment (bhourahine) [inline doc/dftb+/manual/dftbp.tex:5536]:
```suggestion
    faithfully represent the charges for a given excited many-body
```

File: doc/dftb+/manual/manual.bib [hunk 6]
```diff
@@ -1135,3 +1135,12 @@ @ARTICLE{Harabuchi2019
   publisher = {Elsevier}
 }
 
+@article{plotz2014new,
+  title={A new efficient method for calculation of Frenkel exciton parameters in molecular aggregates},
```
Comment (bhourahine) [inline doc/dftb+/manual/manual.bib:1139]:
```suggestion
  title={A new efficient method for calculation of {F}renkel exciton parameters in molecular aggregates},
```

File: src/dftbp/timedep/linrespcommon.F90 [hunk 7]
```diff
@@ -1687,7 +1690,7 @@ subroutine writeExcMulliken(sym, nstat, dq, dqex, coord0)
 
     !> Change in atomic charges from ground to excited state
     real(dp), intent(in) :: dqex(:)
-
+
```
Comment (bhourahine) [inline src/dftbp/timedep/linrespcommon.F90:1693]:
```suggestion

```

File: src/dftbp/timedep/linrespcommon.F90 [hunk 8]
```diff
@@ -1709,7 +1712,7 @@ subroutine writeExcMulliken(sym, nstat, dq, dqex, coord0)
     do m = 1, natom
       write(fdMulliken%unit,"(i5,1x,f16.8)") m, -dq(m) - dqex(m)
     end do
-    call closeFile(fdMulliken)
+    call closeFile(fdMulliken)
```
Comment (bhourahine) [inline src/dftbp/timedep/linrespcommon.F90:1715]:
```suggestion
    call closeFile(fdMulliken)
```

File: src/dftbp/dftbplus/parser.F90
Comment (bhourahine) [inline src/dftbp/dftbplus/parser.F90:1]: Any reason to change this file?

File: src/dftbp/timedep/linrespgrad.F90 [hunk 9]
```diff
@@ -567,8 +568,8 @@ subroutine LinRespGrad_old(env, this, denseDesc, grndEigVecs, grndEigVal, sccCal
     if (this%writeTrans) then
       call openFile(fdTrans, transitionsOut, mode="w")
       write(fdTrans%unit,*)
-    end if
-
+    end if
```
Comment (bhourahine) [inline src/dftbp/timedep/linrespgrad.F90:571]:
```suggestion
    end if
```

File: src/dftbp/timedep/linrespgrad.F90 [hunk 10]
```diff
@@ -567,8 +568,8 @@ subroutine LinRespGrad_old(env, this, denseDesc, grndEigVecs, grndEigVal, sccCal
     if (this%writeTrans) then
       call openFile(fdTrans, transitionsOut, mode="w")
       write(fdTrans%unit,*)
-    end if
-
+    end if  
+
```
Comment (bhourahine) [inline src/dftbp/timedep/linrespgrad.F90:572]:
```suggestion

```

**Review by bhourahine — APPROVED** [review]
Other than checking whether there are MPI fiile issues, and a couple of minor suggestions, looks OK.

File: doc/dftb+/manual/dftbp.tex (hunk 2 above)
Comment (thomas-niehaus) [reply doc/dftb+/manual/dftbp.tex:5528]: Transition is incorrect, as charges are for a many-body state and not transition charges.

**Comment by bhourahine** [comment]
@thomas-niehaus Could you squash this down to one commit and update for the main branch. There were some changes in the main code to protect parallel file writing, so there might be some new warnings for the file if opened on multiple ranks.

**Comment by thomas-niehaus** [comment]
> @thomas-niehaus Could you squash this down to one commit and update for the main branch. There were some changes in the main code to protect parallel file writing, so there might be some new warnings for the file if opened on multiple ranks.

I would need help with this rebase. I tried to rebase as 'git rebase -i HEAD~13' and I got merge conflicts. 
Auto-merging src/dftbp/timedep/linrespgrad.F90
CONFLICT (content): Merge conflict in src/dftbp/timedep/linrespgrad.F90
error: could not apply 5fd54e21... Fix Z-vector timer in LR-TD-DFTB module (#1514)
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 5fd54e21... Fix Z-vector timer in LR-TD-DFTB module (#1514)

How can there be merge conflicts?

**Comment by thomas-niehaus** [comment]
> > @thomas-niehaus Could you squash this down to one commit and update for the main branch. There were some changes in the main code to protect parallel file writing, so there might be some new warnings for the file if opened on multiple ranks.
> 
> I would need help with this rebase. I tried to rebase as 'git rebase -i HEAD~13' and I got merge conflicts. Auto-merging src/dftbp/timedep/linrespgrad.F90 CONFLICT (content): Merge conflict in src/dftbp/timedep/linrespgrad.F90 error: could not apply [5fd54e2](https://github.com/dftbplus/dftbplus/commit/5fd54e21b0725f9c5a7dad12e07f8a965906a654)... Fix Z-vector timer in LR-TD-DFTB module (#1514) Resolve all conflicts manually, mark them as resolved with "git add/rm <conflicted_files>", then run "git rebase --continue". You can instead skip this commit: run "git rebase --skip". To abort and get back to the state before "git rebase", run "git rebase --abort". Could not apply [5fd54e2](https://github.com/dftbplus/dftbplus/commit/5fd54e21b0725f9c5a7dad12e07f8a965906a654)... Fix Z-vector timer in LR-TD-DFTB module (#1514)
> 
> How can there be merge conflicts?

Or should I have rebased first and then merged main? But then I get an additional commit. 

How to proceed from this point?

**Comment by bhourahine** [comment]
@thomas-niehaus Now squashed to a single commit and based on current main. Note the commit ID and date has changed.

**Comment by thomas-niehaus** [comment]
Thanks Ben for helping me out!

**Pull request merged**